--- a/mypyc/build.py
+++ b/mypyc/build.py
@@ -507,6 +507,7 @@ def mypycify(
             '-O{}'.format(opt_level), '-Werror', '-Wno-unused-function', '-Wno-unused-label',
             '-Wno-unreachable-code', '-Wno-unused-variable',
             '-Wno-unused-command-line-argument', '-Wno-unknown-warning-option',
+            '-Wno-unused-but-set-variable',
         ]
         if 'gcc' in compiler.compiler[0]:
             # This flag is needed for gcc but does not exist on clang.
